Build failure: pkgsStatic.hdrhistogram_c

### Nixpkgs version

- Unstable (26.11)

### Steps to reproduce

run `nix build github:NixOS/nixpkgs/234eb9d49b6d9c591c5aa40a55b30eea653e711b#pkgsStatic.hdrhistogram_c`

### Can Hydra reproduce this build failure?

Hydra doesn’t try to build the package.

### Link to Hydra build job

_No response_

### Relevant log output

```console
[  5%] Building C object src/CMakeFiles/hdr_histogram.dir/hdr_histogram_log.c.o
[  5%] Building C object src/CMakeFiles/hdr_histogram.dir/hdr_thread.c.o
[ 10%] Building C object src/CMakeFiles/hdr_histogram.dir/hdr_encoding.c.o
[ 13%] Building C object src/CMakeFiles/hdr_histogram_static.dir/hdr_encoding.c.o
[ 13%] Building C object src/CMakeFiles/hdr_histogram.dir/hdr_histogram.c.o
[ 15%] Building C object src/CMakeFiles/hdr_histogram_static.dir/hdr_histogram.c.o
[ 18%] Building C object src/CMakeFiles/hdr_histogram.dir/hdr_interval_recorder.c.o
[ 21%] Building C object src/CMakeFiles/hdr_histogram_static.dir/hdr_histogram_log.c.o
[ 23%] Building C object src/CMakeFiles/hdr_histogram_static.dir/hdr_interval_recorder.c.o
[ 26%] Building C object src/CMakeFiles/hdr_histogram.dir/hdr_time.c.o
[ 28%] Building C object src/CMakeFiles/hdr_histogram_static.dir/hdr_thread.c.o
[ 31%] Building C object src/CMakeFiles/hdr_histogram.dir/hdr_writer_reader_phaser.c.o
[ 34%] Building C object src/CMakeFiles/hdr_histogram_static.dir/hdr_time.c.o
[ 36%] Building C object src/CMakeFiles/hdr_histogram_static.dir/hdr_writer_reader_phaser.c.o
[ 42%] Linking C shared library libhdr_histogram.so
[ 42%] Linking C static library libhdr_histogram_static.a
/nix/store/9x2vrzrw9p7sfws07zl6a2r6kk8x6b44-x86_64-unknown-linux-musl-binutils-2.46/bin/x86_64-unknown-linux-musl-ld.bfd: /nix/store/f3q65wc4lbbkdvg15am4qwp2bc6gvwx9-x86_64-unknown-linux-musl-gcc-15.3.0/lib/gcc/x86_64-unknown-linux-musl/15.3.0/crtbeginT.o: relocation R_X86_64_32 against hidden symbol `__TMC_END__' can not be used when making a shared object
/nix/store/9x2vrzrw9p7sfws07zl6a2r6kk8x6b44-x86_64-unknown-linux-musl-binutils-2.46/bin/x86_64-unknown-linux-musl-ld.bfd: failed to set dynamic section sizes: bad value
collect2: error: ld returned 1 exit status
make[2]: *** [src/CMakeFiles/hdr_histogram.dir/build.make:197: src/libhdr_histogram.so.6.3.3] Error 1
make[1]: *** [CMakeFiles/Makefile2:1120: src/CMakeFiles/hdr_histogram.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
[ 42%] Built target hdr_histogram_static
make: *** [Makefile:166: all] Error 2
```

### Additional context

nixpkgs-review-gha build job output: https://github.com/applePrincess/nixpkgs-review-gha/actions/runs/31016740240

### System metadata

 - system: `"x86_64-linux"`
 - host os: `Linux 7.1.3, NixOS, 26.11 (Zokor), 26.11.20260705.2ef54b3`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.35pre20260619_f8bb823a`
 - channels(root): `"nixos"`
 - channels(alice): `""`
 - nixpkgs: `/nix/store/kzcva4qkmny8dphvjk92l87wd8al22ga-source`


### Notify maintainers


@jherland 

---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+label%3A%220.kind%3A+build+failure%22).
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.
- [x] I assert that I have read the [automation/AI policy](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#automationai-policy) and that this issue report complies with it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/by-name/hd/hdrhistogram_c/package.nix
+++ b/pkgs/by-name/hd/hdrhistogram_c/package.nix
@@ -32,6 +32,20 @@ stdenv.mkDerivation (finalAttrs: {
     validatePkgConfig
   ];
 
+  cmakeFlags = lib.optionals stdenv.hostPlatform.isStatic [
+    (lib.cmakeBool "HDR_HISTOGRAM_BUILD_SHARED" false)
+    # Examples and tests depend on the shared library target; skip them in
+    # static builds (tests still run for the regular pkgs.hdrhistogram_c build).
+    (lib.cmakeBool "HDR_HISTOGRAM_BUILD_PROGRAMS" false)
+  ];
+
+  # The .pc file always references -lhdr_histogram, but in static builds only
+  # libhdr_histogram_static.a is produced. Provide a symlink so pkg-config
+  # consumers find the right archive.
+  postInstall = lib.optionalString stdenv.hostPlatform.isStatic ''
+    ln -s $out/lib/libhdr_histogram_static.a $out/lib/libhdr_histogram.a
+  '';
+
   doCheck = true;
 
   passthru = {

